ignite/engine/engine.py

```diff
@@ -748,8 +766,11 @@ def _internal_run(self) -> State:
                 handlers_start_time = time.time()
                 if self.should_terminate:
                     self._fire_event(Events.TERMINATE)
-                else:
+                elif self.state.iteration % self.state.epoch_length == 0:
```

Comment:
This can lead to bug if `self.state.epoch_length` is None somehow here.
I would keep the previous code. Especially, as we set should terminate to True below
```python
                if self.state.max_iters is not None and self.state.iteration == self.state.max_iters:
                    self.should_terminate = True
                    break
```